# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/arm/Lowering-arm.cpp
# Commit: 444f4554ea94
# Full Hash: 444f4554ea9456909908f2872d1359495d9ff8d3
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-09-11 15:10:42
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 6.2: Establish dependency from tls for x86 callWithABI div/mod i64. r=lth
#   
#   x86 has few register so to do div/mod for i64 it call the runtime
#   and clobber almost all gp registers including WasmTlsReg.
#   To be able to call c++ runtime via Builtin thunk we need to set up WasmTlsReg.
# ==============================================================================

diff -r 5dd737d76ead -r 444f4554ea94 js/src/jit/arm/Lowering-arm.cpp
--- a/js/src/jit/arm/Lowering-arm.cpp	Fri Sep 11 09:14:07 2020 +0000
+++ b/js/src/jit/arm/Lowering-arm.cpp	Fri Sep 11 09:30:56 2020 +0000
@@ -419,6 +419,10 @@
   defineReturn(lir, div);
 }
 
+void LIRGeneratorARM::lowerWasmBuiltinDivI64(MWasmBuiltinDivI64* div) {
+  MOZ_CRASH("We don't use runtime div for this architecture");
+}
+
 void LIRGeneratorARM::lowerModI64(MMod* mod) {
   if (mod->isUnsigned()) {
     lowerUModI64(mod);
@@ -430,6 +434,10 @@
   defineReturn(lir, mod);
 }
 
+void LIRGeneratorARM::lowerWasmBuiltinModI64(MWasmBuiltinModI64* mod) {
+  MOZ_CRASH("We don't use runtime mod for this architecture");
+}
+
 void LIRGeneratorARM::lowerUDivI64(MDiv* div) {
   LUDivOrModI64* lir = new (alloc()) LUDivOrModI64(
       useInt64RegisterAtStart(div->lhs()), useInt64RegisterAtStart(div->rhs()));